flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -80,12 +95,27 @@ AliasResult AliasAnalysis::alias(Value lhs, Value rhs) {
              llvm::dbgs() << "  rhsSrc: " << rhsSrc << "\n";
              llvm::dbgs() << "\n";);
 
+  // SourceKind::Unknown is set for the addresses wrapped in a global boxes.
```

Comment:
I agree that there is a bit of overlap between these unknowns and the general unknowns. If things are truly unknown, how can we know that they are not target or pointer? I had started up with the introduction of a new SourceKind but early feedback deemed it too complex. However from this baseline, I believe it would be reasonable to reintroduce it.